OpenRA.Mods.Common/UtilityCommands/CheckImageReferences.cs

```diff
@@ -41,6 +41,25 @@ void IUtilityCommand.Run(Utility utility, string[] args)
 				var nodes = MiniYaml.Merge(modData.Manifest.Sequences.Select(s => MiniYaml.FromStream(modData.DefaultFileSystem.Open(s), s)));
 				foreach (var n in nodes.Where(node => !node.Key.StartsWith(ActorInfo.AbstractActorPrefix, StringComparison.Ordinal)))
 					modData.SpriteSequenceLoader.ParseSequences(modData, ts, sc, n);
+
+				foreach (var template in ts.Templates)
+				{
+					foreach (var image in template.Value.Images)
+					{
+						if (!modData.DefaultFileSystem.Exists(image))
+						{
+							failed = true;
+							Console.WriteLine("\t" + "Tileset file {0} not found.".F(image));
+						}
+
+						var sprites = sc[image];
+						if (template.Value.TilesCount > sprites.Length)
+						{
+							failed = true;
+							Console.WriteLine("\t" + "Tileset tile {0} has frames defined without matching artwork.".F(template.Key));
```

Comment:
It seems strange to me that we would crash with a useful exception in this case, but silently continue without crashing if the whole sprite is missing. We could potentially follow the lead of `modData.SpriteSequenceLoader.OnMissingSpriteError` for tilesets too?

Reply:
Can you elaborate on that? This check worked fine for me although my mod only has one tile set.

Reply:
1. Edit w1.sno in RA's `snow.yaml` to `fail.sno`. This crashes the utility, but is easily fixed by adding a `continue` after the log on line 52 and the command then works as expected

2. Undo that edit and instead add a new tile `1: Water`. This crashes the utility before it can even check the sequences, because of checks inside `TileSet` itself, so the code here never actually runs.

Reply:
Went with `modData.SpriteSequenceLoader.OnMissingSpriteError` for tilesets, too now.